internal/services/groups/groups_data_source.go

```diff
@@ -102,7 +109,10 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 			}
 
 			count := len(*result)
-			if count > 1 {
+
+			if securityEnabled {
+				groups = append(groups, *result...)
+			} else if count > 1 {
```

Comment:
This looks like it will just append the results without checking the security-enabled status of the discovered group?

Reply:
This has been changed and the logic moved into the results fetching